### PR TITLE
Allow up-to 6 decimals for transfer values

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Allow up-to 6 decimals for transfer values
- Apply gas cost flooring up-to 6 decimals

Related issues:
- https://github.com/ainblockchain/ain-blockchain/issues/1208

### CODE REVIEW

File: db/bandage-files/allow_up_to_6_decimal_transfer_value_only.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        }
+      },
+      // From allow_non_negative_transfer_value_only.js bandage file.
+      prevValue: {
+        ".rule": {
+          "write": "(auth.addr === $from || auth.fid === '_stake' || auth.fid === '_unstake' || auth.fid === '_pay' || auth.fid === '_claim' || auth.fid === '_hold' || auth.fid === '_release' || auth.fid === '_collectFee' || auth.fid === '_claimReward' || auth.fid === '_openCheckout' || auth.fid === '_closeCheckout' || auth.fid === '_closeCheckin') && !getValue('transfer/' + $from + '/' + $to + '/' + $key) && (util.isServAcntName($from, blockNumber) || util.isCksumAddr($from)) && (util.isServAcntName($to, blockNumber) || util.isCksumAddr($to)) && $from !== $to && util.isNumber(newData) && newData > 0 && util.getBalance($from, getValue) >= newData"
```
Comment (jiyoung-an) [inline db/bandage-files/allow_up_to_6_decimal_transfer_value_only.js:13]: What is a service account name, and how does it differ from a checksum address?

File: blockchain-configs/base/timer_flags.json [hunk 2]
```diff
@@ -55,5 +55,9 @@
   "tweak_transfer_gc_rule": {
     "enabled_block": 2,
     "has_bandage": true
+  },
+  "allow_up_to_6_decimal_transfer_value_only": {
```
Comment (jiyoung-an) [inline blockchain-configs/base/timer_flags.json:59]: What is `timer_flag`?

File: db/bandage-files/allow_up_to_6_decimal_transfer_value_only.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
+  data: [
+    {
+      path: ['rules', 'transfer', '$from', '$to', '$key', 'value' ],
+      value: {
+        '.rule': {
+          "write": "(auth.addr === $from || auth.fid === '_stake' || auth.fid === '_unstake' || auth.fid === '_pay' || auth.fid === '_claim' || auth.fid === '_hold' || auth.fid === '_release' || auth.fid === '_collectFee' || auth.fid === '_claimReward' || auth.fid === '_openCheckout' || auth.fid === '_closeCheckout' || auth.fid === '_closeCheckin') && !getValue('transfer/' + $from + '/' + $to + '/' + $key) && (util.isServAcntName($from, blockNumber) || util.isCksumAddr($from)) && (util.isServAcntName($to, blockNumber) || util.isCksumAddr($to)) && $from !== $to && util.isNumber(newData) && newData > 0 && util.countDecimals(newData) <= 6 && util.getBalance($from, getValue) >= newData"
```
[the remark this answers is not in the document]
Comment (platfowner) [reply db/bandage-files/allow_up_to_6_decimal_transfer_value_only.js:7]: When a native function is triggered, it runs with only its auth.fid set clearing the original user's auth.addr. This distinguishes state writes by native functions from those by users.

File: db/bandage-files/allow_up_to_6_decimal_transfer_value_only.js (hunk 1 above)
Comment (platfowner) [reply db/bandage-files/allow_up_to_6_decimal_transfer_value_only.js:13]: Service accounts are accounts that can hold tokens for service constructions. For example, the accounts under /service_accounts/staking/consensus are service accounts for staking to participate in the blockchain consensus (block producing).
https://testnet-insight.ainetwork.ai/database/values/service_accounts/staking/consensus

File: blockchain-configs/base/timer_flags.json (hunk 2 above)
Comment (platfowner) [reply blockchain-configs/base/timer_flags.json:59]: Timer flags are a flag with its enabled and disabled block numbers. For backward compatibility, any new features that affects the consensus (including state root hash) need to be enabled after specific block number.

**Comment by platfowner** [comment]
Thanks for the review! @jiyoung-an Now merging..

**Pull request merged**